packages/flutter/lib/src/material/button_style_button.dart

```diff
@@ -11,11 +11,43 @@ import 'package:flutter/widgets.dart';
 import 'button_style.dart';
 import 'colors.dart';
 import 'constants.dart';
+import 'elevated_button.dart';
+import 'filled_button.dart';
 import 'ink_well.dart';
 import 'material.dart';
 import 'material_state.dart';
+import 'outlined_button.dart';
+import 'text_button.dart';
 import 'theme_data.dart';
 
+/// {@template flutter.material.ButtonStyleButton.iconAlignment}
+/// Determines the alignment of the icon within the widgets such as:
+///   - [TextButton.icon],
+///   - [ElevatedButton.icon],
+///   - [OutlinedButton.icon],
+///   - [FilledButton.icon],
+///   - [FilledButton.tonalIcon].
+///
+/// The effect of `iconAlignment` depends on [TextDirection]. If textDirection is [TextDirection.ltr] then [IconAlignment.start] and [IconAlignment.end] align the icon on the left or right respectively.  If textDirection is [TextDirection.rtl] then the alignments are reversed.
+///   - If textDirection is [TextDirection.ltr] & iconAlignment is [IconAlignment.start],
+///     it aligns the icon at the start of the button.
+///   - If textDirection is [TextDirection.rtl] & iconAlignment is [IconAlignment.start],
+///     it aligns the icon at the end of the button.
+///   - If textDirection is [TextDirection.ltr] & iconAlignment is [IconAlignment.end],
+///     it aligns the icon at the end of the button.
+///   - If textDirection is [TextDirection.rtl] & iconAlignment is [IconAlignment.end],
+///     it aligns the icon at the start of the button.
```

Comment:
This is what I was suggesting for the doc change:

```suggestion
/// The effect of `iconAlignment` depends on [TextDirection]. If textDirection is
/// [TextDirection.ltr] then [IconAlignment.start] and [IconAlignment.end] align the
/// icon on the left or right respectively.  If textDirection is [TextDirection.rtl] the
/// the alignments are reversed.
```

Reply:
my bad, thanks!